src/main/java/core/basesyntax/strategy/impl/AbstractOperationHandler.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.enums.Operation;
+import core.basesyntax.strategy.OperationHandler;
+import java.util.Map;
+
+public abstract class AbstractOperationHandler implements OperationHandler {
+    private final Operation targetOperation;
+
+    protected AbstractOperationHandler(Operation targetOperation) {
+        this.targetOperation = targetOperation;
+    }
+
+    @Override
+    public void process(Map<String, Integer> fruitQuantityMap, FruitTransaction transaction) {
+        validateTransactionOperation(transaction);
+        processTransaction(fruitQuantityMap, transaction);
+    }
+
+    protected abstract void processTransaction(Map<String, Integer> fruitQuantityMap,
+                                               FruitTransaction transaction);
+
+    void validateTransactionOperation(FruitTransaction transaction) {
```

Comment:
According to the encapsulation principle, the visibility of `validateTransactionOperation` should be restricted. It is not meant to be used outside of the hierarchy of classes that extend `AbstractOperationHandler`. Consider making it `protected` to encapsulate it within the package and subclasses.